src/Context.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useMemo, useState } from 'react';
+import { Post } from './types/interfaces';
+
+interface IAppContext {
+  isModalWindowShown: boolean;
```

Comment:
Let's think about all needed items for context
1) You need to have just `modaltype` property. It could be one of type of modal content or null if modal should be closed
2) toggleModalType - method for changing the modal type
3) You have 2 ways: 

- create additional property for storing data (in your case it's edited post data) but try to avoid using `editedPost` or some specific props for this, let's make it generic and use something like `modalData` or `state` or `modalState`
- or extend modalType  and make it as object with `type` and `state`